### PR TITLE
ISD-Argo support added

### PR DESCRIPTION
**Pull request by Pranav-b-7**
https://devopsmx.atlassian.net/browse/OP-19540

### CODE REVIEW

File: opsmx-isd-register/src/main/java/com/opsmx/isd/register/interceptors/ISDSaasTrialInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+public class ISDSaasTrialInterceptor implements HandlerInterceptor {
+
+    @Value("${api.key}")
+    String apiKey;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+
+        if(Boolean.FALSE.equals(Util.rateLimit(request))){
+            throw new TooManyRequestsException("You have exceeded the 10 requests in 1 minute limit!");
```
Comment (sriharshakancharla) [inline opsmx-isd-register/src/main/java/com/opsmx/isd/register/interceptors/ISDSaasTrialInterceptor.java:25]: Please add the standard error message format

File: opsmx-isd-register/src/main/java/com/opsmx/isd/register/interceptors/ISDSaasTrialInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        }
+
+        String xApiKey = request.getHeader("x-api-key");
+        if (xApiKey!=null && xApiKey.trim().equalsIgnoreCase(apiKey)){
+
+            return Boolean.TRUE;
+        }
+        throw new AccessForbiddenException("You do not have enough privileges to access this resource");
```
Comment (sriharshakancharla) [inline opsmx-isd-register/src/main/java/com/opsmx/isd/register/interceptors/ISDSaasTrialInterceptor.java:33]: Please add the standard error message format

File: opsmx-isd-register/src/main/java/com/opsmx/isd/register/service/AccountSetupServiceImpl.java [hunk 3]
```diff
@@ -57,25 +72,39 @@ public DatasourceResponseModel setup(DatasourceRequestModel datasourceRequestMod
...
+
+    private DatasourceResponseModel getDatasourceResponseModel(DatasourceResponseModel datasourceResponseModel, ResponseEntity<DatasourceResponseModel> responseEntity) {
+        try {
+            DatasourceResponseModel responseModel = responseEntity.getBody();
+            if(responseModel != null && responseModel.getEventProcessed()){
+                log.info("Event trigger ISD register success, event ID : {}", responseModel.getEventId());
+                return responseModel;
+            }
+        } catch (Exception e) {
+            log.error("Exception in triggering ISD register event {}", e);
```
Comment (sriharshakancharla) [inline opsmx-isd-register/src/main/java/com/opsmx/isd/register/service/AccountSetupServiceImpl.java:94]: add print stack trace also

File: opsmx-isd-register/src/main/java/com/opsmx/isd/register/interceptors/ISDSaasTrialInterceptor.java (hunk 1 above)
Comment (Pranav-b-7) [reply opsmx-isd-register/src/main/java/com/opsmx/isd/register/interceptors/ISDSaasTrialInterceptor.java:25]: Defining std error codes wont be necessary at the moment as there are very minimal error handling logics here . And moreover its a refactoring , so we shall handle it later on

File: opsmx-isd-register/src/main/java/com/opsmx/isd/register/service/AccountSetupServiceImpl.java (hunk 3 above)
Comment (Pranav-b-7) [reply opsmx-isd-register/src/main/java/com/opsmx/isd/register/service/AccountSetupServiceImpl.java:94]: just specifying the variable itself would print the stack trace

**Review by sriharshakancharla — APPROVED** [review]

**Pull request merged**